### PR TITLE
General Terms of Use

### PR DESCRIPTION
**Pull request by shikokuchuo**
Closes https://github.com/r-multiverse/help/issues/82.

Designed to apply to all Users, protecting Admins / Moderators / Contributors (collectively the 'Contributors').

The legal jurisdiction is still in [ ]s. I am comfortable with using the law of England and Wales as it operates under a case law system with ample (commercial) precedent, hence predictability. But I'd be open to researching the implications of using US law etc. if that is to be preferred.

Comments welcome.

### CODE REVIEW

File: terms.md
Comment (maelle) [inline terms.md:7]: I'd recommend line breaks after each sentence for better future diff-ability.

File: terms.md
Comment (maelle) [inline terms.md:13]: should the link be to a rendered version not the md?

File: terms.md
Comment (maelle) [inline terms.md:33]: I am not knowledgeable at all in law so can't comment on this.

File: terms.md
Comment (maelle) [inline terms.md:37]: how would we advertise the changes? This sentence reminds me of Anaconda :sweat_smile:

File: terms.md
Comment (maelle) [inline terms.md:41]: access to developers?

**Review by maelle — COMMENTED** [review]
thank you!

File: terms.md
Comment (shikokuchuo) [reply terms.md:13]: Quarto allows this, so the rendered version will link to the html. I don't know what is best practice here.

File: terms.md
Comment (shikokuchuo) [reply terms.md:33]: OK, thanks. I'll leave this point open until @wlandau is back. Just in case he wants to pass this through Lilly legal.

File: terms.md
Comment (shikokuchuo) [reply terms.md:37]: Ah, let me tie this back in to the Governance Document with the one month notice period. I think the more important point is the following one about termination of access at any time.

File: terms.md
Comment (shikokuchuo) [reply terms.md:41]: Access to users, whoever they may be. The wording here is designed to protect us in situations such as we need to take the site offline due to security concerns, or I'd say even more broadly if there is an outage (outside our control), that there is no presumptive right to continued access at any time.

File: terms.md
Comment (maelle) [reply terms.md:13]: oooh that's awesome

File: terms.md
Comment (shikokuchuo) [reply terms.md]: Thanks for reminding!

File: terms.md
Comment (shikokuchuo) [reply terms.md]: Amended in [f13aa58](https://github.com/r-multiverse/r-multiverse.github.io/pull/25/commits/f13aa58585598a16d679b5b9befc4b5763301cfb).

File: terms.md
Comment (shikokuchuo) [reply terms.md]: Actualy I'll resolve this and add a comment at the top level to make it easier for Will to find.

**Comment by shikokuchuo** [comment]
@wlandau the proposed legal jurisdiction is that of England and Wales. This is generally fine for similar contracts where operations are global. Please let us know if this is an issue for you.

File: terms.md
Comment (wlandau) [inline terms.md:17]: > any Administrator or Moderator of the Repository, nor any Contributor to the Repository (collectively, "the Contributors")

I think "the Contributors" might be ambiguous with the way "Contributor" is defined in https://r-multiverse.org/governance.html#contributor. How about "Developers"?

I would recommend a Definitions section which defines Developer (admins, moderators, contributors), Maintainer (package maintainer), and User.

File: terms.md
Comment (wlandau) [inline terms.md:28]: Maybe just "Violation of these Terms" in general? E.g. if a contributor posts malware that harms a user. Likewise with the other points.

File: terms.md
Comment (wlandau) [reply terms.md:17]: We might need to also cover the definitions in https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#a-definitions.

File: terms.md
Comment (wlandau) [inline terms.md:10]: What about an "Acceptable use" section for users? From https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#c-acceptable-use:

> Your use of the Website and Service must not violate any applicable laws, including copyright or trademark laws, export control or sanctions laws, or other laws in your jurisdiction. You are responsible for making sure that your use of the Service is in compliance with laws and any applicable regulations.

For example, what if a user is legally prohibited from downloading a package in their country even if we think the package itself is okay?

File: terms.md
Comment (wlandau) [inline terms.md:13]: This is an important section, and I recommend subsuming all the relevant parts we can glean from https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#n-disclaimer-of-warranties.

File: terms.md
Comment (wlandau) [inline terms.md:21]: We might borrow from https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#o-limitation-of-liability to expand this section.

File: terms.md
Comment (wlandau) [reply terms.md:28]: We'll want to cover ourselves against all the relevant stuff listed at https://docs.github.com/en/site-policy/acceptable-use-policies. More on that in a general comment...

**Review by wlandau — CHANGES_REQUESTED** [review]
Great start, @shikokuchuo! 

As you may have noticed, my comments borrow a lot from https://docs.github.com/en/site-policy/github-terms/github-terms-of-service.

This Terms of Use document seems to cover "Users" as defined at https://r-multiverse.org/governance.html#user. I think we will somehow want to cover Contributors and potentially package maintainers as well. https://docs.github.com/en/site-policy/acceptable-use-policies has a lot of great stuff, and it covers some of the scariest failure modes. I'm thinking we could consider subsuming all that stuff under "Terms of Use" and potentially even supersede our current code of conduct.

File: terms.md
Comment (shikokuchuo) [reply terms.md]: I prefer not to also have a definitions section as the style currently employed is more direct:

> These Terms of Use (the "Terms") govern your use of R-multiverse (the "Repository").
By using the Repository, you agree to comply with these Terms.
If you do not agree with these Terms, you may not use the Repository. 

I'm comfortable with using plain language without needing to be too 'legalistic'. Contradictions are more painful to deal with.

To remove ambiguity, I've reworded to use "Project Participants" as a neutral-sounding term.

File: terms.md
Comment (shikokuchuo) [reply terms.md:28]: > Maybe just "Violation of these Terms" in general? E.g. if a contributor posts malware that harms a user. Likewise with the other points.

To be clear - the intent of this document is to protect everyone involved in the project (Project Participants) from anyone using the site. If we want to address a contributor posting malware then I think that belongs somewhere else e.g. Code of Conduct.

File: terms.md
Comment (shikokuchuo) [reply terms.md:10]: I think we've covered by the general indemnity under the terms, and it is not for us to stipulate what a user does (and it is not reasonable for us to do so).

However, I've added a bullet to the indemnity covering the above, as it's not an exhaustive list anyway.

File: terms.md
Comment (shikokuchuo) [reply terms.md:13]: I've added to this section.

File: terms.md
Comment (shikokuchuo) [reply terms.md]: I've also beefed up this section.

**Comment by shikokuchuo** [comment]
@wlandau do take a look at the changes I've made. Can iterate further if you have other comments.
Also are you happy to take [England and Wales] law out of square brackets?

File: terms.md
Comment (wlandau) [reply terms.md]: > I'm comfortable with using plain language without needing to be too 'legalistic'. 

Plain language is good when possible, but ultimately this is a legal document whose purpose is to provide legal protections. If the document does not carefully define its terms, this could invite deliberately wrong interpretations which favor prosecutors in a lawsuit.

The GitHub terms of service includes [Definitions section](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#a-definitions) while keeping its language concise, digestible, and in plain English.

File: terms.md
Comment (wlandau) [reply terms.md]: >  If we want to address a contributor posting malware then I think that belongs somewhere else e.g. Code of Conduct.

In that case, I suggest we write an Acceptable Use policy and reference it from this one, e.g. 

> You agree to indemnify, defend, and hold harmless the Project Participants from and against any claims, liabilities, damages, losses, and expenses, including legal fees and costs, arising out of or in any way connected with your access to or use of the Repository, including but not limited to:
> 
> * Your violations of these Terms
> * Violations of the Acceptable Use Policy
> * ...

Comment (wlandau) [reply terms.md]: (I don't think we need to reference the Acceptable Use Policy until it actually exists.)

File: terms.md
Comment (wlandau) [reply terms.md:10]: > I think we're covered by the general indemnity under the terms, and it is not for us to stipulate what a user does (and it is not reasonable for us to do so).

Whether in this document or elsewhere, there are other aspects of acceptable use that we need to cover: for example, [violent](https://docs.github.com/en/site-policy/acceptable-use-policies/github-threats-of-violence-and-gratuitously-violent-content) or [obscene](https://docs.github.com/en/site-policy/acceptable-use-policies/github-sexually-obscene-content) content that is illegal to even own. A third-party contributor could upload illegal obscene images, and then they would be in our repository. Even if nobody downloads those images, we could still face prosecution.

We need to make it clear that the package maintainers have ownership and responsibility for content, not us, and that we are indemnified from the actions of contributors and package maintainers. The current document only indemnifies us from damages of people downloading packages.

> However, I've added a bullet to the indemnity covering the above, as it's not an exhaustive list anyway.

Thanks for that.

**Comment by wlandau** [comment]
> Also are you happy to take [England and Wales] law out of square brackets?

That's fine.

> @wlandau do take a look at the changes I've made. Can iterate further if you have other comments.

Thanks for the changes.

From https://github.com/r-multiverse/r-multiverse.github.io/pull/25#discussion_r1790545104, there is a lot to cover under an Acceptable Use document. However, for that to work, I think these Terms needs to make responsibility and ownership clear. I like these paragraphs from [GitHub](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service)'s Terms of Service:

> C. Acceptable Use
>
> Your use of the Website and Service must not violate any applicable laws, including copyright or trademark laws, export control or sanctions laws, or other laws in your jurisdiction. You are responsible for making sure that your use of the Service is in compliance with laws and any applicable regulations.

> D. User-Generated Content
> 1. Responsibility for User-Generated Content
>
> You are solely responsible for the content of, and for any harm resulting from, any User-Generated Content that you post, upload, link to or otherwise make available via the Service, regardless of the form of that Content. We are not responsible for any public display or misuse of your User-Generated Content.

File: terms.md
Comment (shikokuchuo) [reply terms.md]: You missed my point above. I am saying that as long as what we write is in clear and plain English, it is easy to infer intent and the weight of precedent will be such that there is practically no chance of an obscure interpretation winning the day.

On the other hand, if we are make lots of definitions, then these terms are narrowed to precisely what that contains rather than their natural meaning. This introduces risk if we don't define something properly.

So on this point, if you have something specific that you think is unclear, let me know and I'll have a look at it.

If purely as a matter of document format you want a definitions section, that's also fine and I can move the existing definitions up. Let me know.

File: terms.md
Comment (shikokuchuo) [reply terms.md]: OK. If you note, for GitHub it is also a separate document.

File: terms.md
Comment (shikokuchuo) [reply terms.md:10]: This belongs in an Acceptable Use Policy, as you've noted elsewhere as well.

**Comment by shikokuchuo** [comment]
I've taken the jurisdiction out of [ ]s.

The wording of C is already incorporated into this document. As for D, this is applicable for GitHub, but for us the users are not uploading their own content - rather this will be covered for contributors in the Acceptable Use Policy.

File: terms.md
Comment (shikokuchuo) [reply terms.md]: I've also iterated on the key definition, which is who the Project Participants are to make it absolutely clear that protections are to be afforded both jointly and individually in [7719663](https://github.com/r-multiverse/r-multiverse.github.io/pull/25/commits/7719663e23e3ca767f05b46e8e9504ac631b31b7).

Comment (shikokuchuo) [reply terms.md]: @wlandau I've further updated this after drafting the AUP, making the style consistent - bringing the definitions upfront and making them clearer. I believe this is now addressed.

**Comment by shikokuchuo** [comment]
The Acceptable Use Policy is now drafted so these should be read in conjunction with each other.

Due to timing, this document will go into effect before that one - but let me know if anyone has any further comments.

**Comment by wlandau** [comment]
I like the added definitions. I would add a couple more. GitHub's terms of service defines the audience:

> “The User,” “You,” and “Your” refer to the individual person, company, or organization that has visited or is using the Website or Service; that accesses or uses any part of the Account; or that directs the use of the Account in the performance of its functions. A User must be at least 13 years of age. Special terms may apply for business or government Accounts (See [Section B(5): Additional Terms](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#5-additional-terms)).

It would be also good to define "package". We mean it in terms of "R package" as defined in https://cran.r-project.org/doc/FAQ/R-exts.html.

After those changes, I am fine to merge this.

**Comment by wlandau** [comment]
Another thing, actually: we will have "Terms of Use" and "Acceptable Use", which sound similar because of the word "Use". Maybe this one can be "Terms of Service" and define "Service" in the definitions section?

**Comment by shikokuchuo** [comment]
> I like the added definitions. I would add a couple more. GitHub's terms of service defines the audience:
> 
> > “The User,” “You,” and “Your” refer to the individual person, company, or organization that has visited or is using the Website or Service; that accesses or uses any part of the Account; or that directs the use of the Account in the performance of its functions. A User must be at least 13 years of age. Special terms may apply for business or government Accounts (See [Section B(5): Additional Terms](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#5-additional-terms)).

I've added a definition for "you" and "your" that works for us.

> It would be also good to define "package". We mean it in terms of "R package" as defined in https://cran.r-project.org/doc/FAQ/R-exts.html.

I think the term is unambiguous in the context of a repository. I do not want to define it too narrowly here, as in any case we tend to use "packages and services" together.

> 
> After those changes, I am fine to merge this.

Let me know if you're ok with the above.

**Comment by shikokuchuo** [comment]
> Another thing, actually: we will have "Terms of Use" and "Acceptable Use", which sound similar because of the word "Use". Maybe this one can be "Terms of Service" and define "Service" in the definitions section?

Terms of Use is probably more accurate as Term of Service is more geared towards sites with user accounts that are providing a service. For us, it's mostly people using the repository to download packages.

**Comment by wlandau** [comment]
> I've added a definition for "you" and "your" that works for us.

Thanks. I also like the list format you use.

> I think the term is unambiguous in the context of a repository. I do not want to define it too narrowly here, as in any case we tend to use "packages and services" together.

Would this be broad enough?

```
* Service: any capability that R-multiverse provides for you, including but not limited to the ability to download and install packages.
* Package: a collection of code or software, including but not limited to an R extension as described at https://cran.r-project.org/doc/FAQ/R-exts.html.
```

**Comment by wlandau** [comment]
> If you like "Terms of Use", what about "Acceptable Contribution" for the other one?

On second thought, "Terms of Service" is narrower than "Terms of Use" because the latter covers more than just services. Likewise, "Use" is more general than "Contribution". So I am fine with the current names.

**Comment by shikokuchuo** [comment]
> 
> Would this be broad enough?
> 
> ```
> * Service: any capability that R-multiverse provides for you, including but not limited to the ability to download and install packages.
> * Package: a collection of code or software, including but not limited to an R extension as described at https://cran.r-project.org/doc/FAQ/R-exts.html.
> ```

Please see [4affc31](https://github.com/r-multiverse/r-multiverse.github.io/pull/25/commits/4affc3100ef8948a05defe6d5395b6a831048c81) where I've added definitions along those lines. The one for package must be self-contained otherwise it incorporates the linked document by reference and that's never a good idea.

Please double-check where I've capitalised all mentions of "Package" and "Services" that you think the definition is fine in each respect. Thanks.

**Comment by shikokuchuo** [comment]
Added a correction in [8f4ca7e](https://github.com/r-multiverse/r-multiverse.github.io/pull/25/commits/8f4ca7ec78a84522966dfa5e046bedefb3f529b9) - you see how easy it is to get something wrong!

**Review by wlandau — APPROVED** [review]

**Pull request merged**
